pkg/asciisanitizer/sanitizer.go

```diff
@@ -47,7 +47,7 @@ func (t *Sanitizer) Transform(dst, src []byte, atEOF bool) (nDst, nSrc int, err
 			return
 		}
 		r, size := utf8.DecodeRune(src)
-		if r == utf8.RuneError {
+		if r == utf8.RuneError && size == 1 {
```

Comment:
```suggestion
		if r == utf8.RuneError && size < 2 {
```

If the size is 0 or 1 there is an issue so lets check both cases.